### PR TITLE
provider/azurerm: Bump azure-sdk-for-go to 3.0.0-beta

### PR DESCRIPTION
**Pull request by stack72**
There were breaking changes to CDN Endpoint, CDN Profile and Storage
Account

The test results of the changes are as follows:

```
make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMCdnProfile'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /vendor/)
TF_ACC=1 go test ./builtin/providers/azurerm -v
-run=TestAccAzureRMCdnProfile -timeout 120m
=== RUN   TestAccAzureRMCdnProfile_basic
--- PASS: TestAccAzureRMCdnProfile_basic (199.61s)
=== RUN   TestAccAzureRMCdnProfile_withTags
--- PASS: TestAccAzureRMCdnProfile_withTags (214.24s)
PASS
ok      github.com/hashicorp/terraform/builtin/providers/azurerm
413.855s
```

```
make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMCdnEndpoint'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /vendor/)
TF_ACC=1 go test ./builtin/providers/azurerm -v
-run=TestAccAzureRMCdnEndpoint -timeout 120m
=== RUN   TestAccAzureRMCdnEndpoint_basic
--- PASS: TestAccAzureRMCdnEndpoint_basic (226.85s)
=== RUN   TestAccAzureRMCdnEndpoint_withTags
--- PASS: TestAccAzureRMCdnEndpoint_withTags (269.52s)
PASS
ok      github.com/hashicorp/terraform/builtin/providers/azurerm
496.382s
```

```
make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMStorageAccount_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /vendor/)
TF_ACC=1 go test ./builtin/providers/azurerm -v
-run=TestAccAzureRMStorageAccount_ -timeout 120m
=== RUN   TestAccAzureRMStorageAccount_basic
--- PASS: TestAccAzureRMStorageAccount_basic (163.61s)
PASS
ok      github.com/hashicorp/terraform/builtin/providers/azurerm
163.621s
```

### CODE REVIEW

**Comment by apparentlymart** [comment]
:+1: as long as the tests still pass, which remains to be seen at the point where I'm writing this. :grinning:

**Comment by stack72** [comment]
Discovered breaking changes - so will work on fixing them before the merge!

File: builtin/providers/azurerm/resource_arm_cdn_endpoint.go [hunk 1]
```diff
@@ -270,23 +271,13 @@ func resourceArmCdnEndpointUpdate(d *schema.ResourceData, meta interface{}) erro
...
-	if d.HasChange("origin") {
-		origins, originsErr := expandAzureRmCdnEndpointOrigins(d)
-		if originsErr != nil {
-			return fmt.Errorf("Error Building list of CDN Endpoint Origins: %s", originsErr)
-		}
-		if len(origins) > 0 {
-			properties.Origins = &origins
-		}
-	}
-
```
Comment (phinze) [inline builtin/providers/azurerm/resource_arm_cdn_endpoint.go]: Noting for history that this field is changing from updateable to ForceNew, which is why this block goes away. The field is marked as `ForceNew` above.

**Comment by phinze** [comment]
Okay gave this a look. LGTM 👍

**Pull request merged**